frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js

```diff
@@ -207,14 +207,14 @@ const InsideHearingMainPage = () => {
   }, [transcriptText, setTranscriptText]);
 
   const isDepositionSaved = useMemo(() => {
-    return hearing?.additionalDetails?.witnessDepositions?.find((witness) => witness.uuid === selectedWitness.uuid)?.deposition.length;
+    return Boolean(hearing?.additionalDetails?.witnessDepositions?.find((witness) => witness.uuid === selectedWitness.uuid)?.deposition);
   }, [selectedWitness, hearing]);
 
   const saveWitnessDeposition = () => {
     const updatedHearing = structuredClone(hearing);
     setWitnessModalOpen(true);
-    updatedHearing.additionalDetails = updatedHearing.additionalDetails || {};
-    updatedHearing.additionalDetails.witnessDepositions = updatedHearing.additionalDetails.witnessDepositions || [];
+    updatedHearing.additionalDetails = updatedHearing?.additionalDetails || {};
```

Comment:
`updatedHearing?.additionalDetails` protects from `updatedHearing` being falsy. If so, the left hand side of this statement will fail first anyway, so what is the point of this change ?

Reply:
addressed

Reply:
? won't be needed at all since hearing will be always an object